### PR TITLE
Unify `AstroportXykPool` and `AstroportStableSwapPool`

### LINKED ISSUE #45: Unify `AstroportXykPool` and `AstroportStableSwapPool` into one struct
Keep a `PairType` object in struct and implement different functionality in `simulate_provide_liquidity` using `match`. This gets rid of the need to use feature flags or have two separate contracts when implementing the vault contracts

### PR DESCRIPTION
**Pull request by apollo-pacman**
closes #45

### CODE REVIEW

File: src/implementations/astroport/pool.rs [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+            contract_addr: self.pair_addr.to_string(),
+            msg: to_binary(&QueryMsg::Pool {})?,
+        }))
+    }
+
+    /// Math for LP shares calculation when providing liquidity to an Astroport constant product pool.
+    ///
+    /// Copied from the astroport XYK pool implementation here:
+    /// https://github.com/astroport-fi/astroport-core/blob/7bedc6f27e59ef8b921a0980be9bc30c4aab7459/contracts/pair/src/contract.rs#L297-L434
+    pub fn xyk_simulate_provide_liquidity(
```
Comment (apollo-sturdy) [inline src/implementations/astroport/pool.rs:69]: Why this pub but not stableswap equivalent?

File: src/implementations/astroport/helpers.rs [hunk 2]
```diff
@@ -80,4 +92,170 @@ pub fn checked_u8_mul(a: &U256, b: u8) -> Option<U256> {
...
+}
+
+//
+// ============================================================
+// ====== Helper functions for Stableswap implementation ======
+// ============================================================
+//
+
+pub(crate) const N_COINS: u8 = 2;
+pub const AMP_PRECISION: u64 = 100;
```
Comment (apollo-sturdy) [inline src/implementations/astroport/helpers.rs:104]: Could be private or `pub(crate)`?

File: src/implementations/astroport/helpers.rs [hunk 3]
```diff
@@ -80,4 +92,170 @@ pub fn checked_u8_mul(a: &U256, b: u8) -> Option<U256> {
...
+// Raw query gets all the necessary elements
+pub fn query_pair_config(querier: &QuerierWrapper, pair: &Addr) -> StdResult<Config> {
+    if let Some(res) = querier.query_wasm_raw(pair, b"config".as_slice())? {
+        let res: Config = from_slice(&res)?;
+        Ok(res)
+    } else {
+        Err(cosmwasm_std::StdError::GenericErr {
+            msg: "Raw query failed: config not found on pair address".to_string(),
+        })
+    }
```
Comment (apollo-sturdy) [inline src/implementations/astroport/helpers.rs:240]: Why not `Item.query` as discussed?

**Review by apollo-sturdy — APPROVED** [review]
Approved but could maybe change the config query to simplify

File: src/implementations/astroport/pool.rs (hunk 1 above)
Comment (apollo-pacman) [reply src/implementations/astroport/pool.rs:69]:
```suggestion
    fn xyk_simulate_provide_liquidity(
```

**Pull request merged**